frontend/src/metabase/visualizations/components/EChartsRenderer/ResponsiveEChartsRenderer.tsx

```diff
@@ -1,16 +1,42 @@
+import { useEffect } from "react";
 import ExplicitSize from "metabase/components/ExplicitSize";
 import type { EChartsRendererProps } from "metabase/visualizations/components/EChartsRenderer/EChartsRenderer";
 import { EChartsRenderer } from "metabase/visualizations/components/EChartsRenderer/EChartsRenderer";
 import { ResponsiveEChartsRendererStyled } from "metabase/visualizations/components/EChartsRenderer/ResponsiveEChartsRenderer.styled";
 
+interface ResponsiveEChartsRendererProps extends EChartsRendererProps {
+  onResize: (width: number, height: number) => void;
+  width: number;
+  height: number;
+}
+
 export const ResponsiveEChartsRenderer = ExplicitSize({
   wrapped: true,
   refreshMode: "debounceLeading",
   selector: false,
-})((props: EChartsRendererProps) => {
-  return (
-    <ResponsiveEChartsRendererStyled>
-      <EChartsRenderer {...props} />
-    </ResponsiveEChartsRendererStyled>
-  );
-});
+})(
+  ({
+    onResize,
+    width,
+    height,
+    ...echartsRenderedProps
+  }: ResponsiveEChartsRendererProps) => {
+    useEffect(() => {
+      onResize(width, height);
```

Comment:
To correctly compute option we need to use width and height of the chart itself. However, the parent component which is also wrapped in `ExplicitSize` includes legend and paddings so we need to get dimensions from this component